lib/application/view-models/home/collections_vm.dart

```diff
@@ -0,0 +1,173 @@
+import 'package:equatable/equatable.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:memo/domain/models/collection.dart';
+import 'package:meta/meta.dart';
+
+final collectionsVM = StateNotifierProvider<CollectionsVM>((_) => CollectionsVMImpl());
+
+enum CollectionsFilter { explore, review }
+const availableFilters = CollectionsFilter.values;
+
+abstract class CollectionsVM extends StateNotifier<CollectionsState> {
+  CollectionsVM(CollectionsState state) : super(state);
+
+  /// Updates the current [state] with [filter]
+  Future<void> updateCollectionsFilter(CollectionsFilter filter);
```

Comment:
I believe that the `filter` word here is not a good decision. If we add inner filters in `Explorar` and `Revisar` tabs to filter the collections from each tab, these filters naming would conflict heavily with this filter definition. Unless all filters (Explorar & Revisar + inner filters) could be handled as the same thing - and I don't know if it's possible with today's implementation - it would have no problem, but I suspect that this may not be possible with the filter structure.

Reply:
What is your suggestion on this naming? Because that's what they are to the View Model, a filter. If the UI uses them as TABS, that's an UI decision, and not the VM's

Reply:
What about `section`? It indicates that they have a separation but it's not UI specific as `tab` is

Reply:
I can't understand why you see `section` as "UI specific" since it doesn't leak how the UI is structured, just that it has a "major" division. But my point here is not about the `section` thing, I could suggest a list of other synonyms that are worst in my opinion but fit here, like `segment`, `module`, etc... We eventually would find one that it's not UI specific.

But the main point here is about the conflict that may arise if this screen receives a categories filter for each tab, and I don't know if you agree or disagree about it. If you think that this could be handled with two different filter functions like `updateCollectionsFilter` and `updateCategoriesFilter`, we can keep the filter naming, my concern is about it being 
 confuse, that's why I suggest the naming change. But again, you haven't expressed yourself about this possible conflict.